### PR TITLE
Load symbols in modules from PDB 

### PR DESCRIPTION
**Pull request by b1llow**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)
# SQUASH ME

**Detailed description**

...

**Test plan**

...

**Closing issues**

### CODE REVIEW

**Comment by b1llow** [comment]
I tried to fix #955 but I realized that RzType doesn't support cpp class right now and I may not have a simple way to deal with baseclass.

File: librz/include/rz_vector.h [hunk 1]
```diff
@@ -367,11 +367,13 @@ static inline void **rz_pvector_flush(RzPVector *vec) {
  * }
  */
 #define rz_pvector_foreach(vec, it) \
-	for (it = (void **)(vec)->v.a; (vec)->v.len && it != (void **)(vec)->v.a + (vec)->v.len; it++)
+	if (!rz_pvector_empty(vec)) \
```
Comment (XVilka) [inline librz/include/rz_vector.h]: Could you please send this change as a separate PR?

File: librz/core/cpdb.c [hunk 2]
```diff
@@ -114,153 +114,188 @@ static void rz_core_bin_pdb_types_print(const RzTypeDB *db, const RzPdb *pdb, co
...
- * \param mode RzOutputMode
+ * \param baddr image base addr
+ * \param state The RzCmdStateOutput instance
  * \return char *
  */
-RZ_API char *rz_core_bin_pdb_gvars_as_string(RZ_NONNULL const RzPdb *pdb, const ut64 img_base, PJ *pj, const RzOutputMode mode) {
-	rz_return_val_if_fail(pdb, NULL);
-	RzPdbGDataStream *gsym_data_stream = 0;
-	RzPdbPeStream *pe_stream = 0;
+RZ_API char *rz_core_bin_pdb_gvars_as_string(
```
Comment (XVilka) [inline librz/core/cpdb.c:173]: Good catch!

File: librz/bin/pdb/symbol.c [hunk 3]
```diff
@@ -25,7 +22,7 @@ RZ_IPI bool PDBSectionOffset_parse(RzBuffer *b, PDBSectionOffset *section_offset
 
 static bool symbol_name_parse(RzBuffer *b, PDBSymbolKind kind, char **result) {
 	if (kind < S_ST_MAX) {
-		// TODO: pascal
+		return buf_read_u8_pascal_string(b, result);
```
Comment (XVilka) [inline librz/bin/pdb/symbol.c:25]: Do we have a file with this case?

**Review by XVilka — CHANGES_REQUESTED** [review]
Is there any new (or old) test that would check this module parsing?

File: librz/include/rz_vector.h (hunk 1 above)
Comment (b1llow) [reply librz/include/rz_vector.h]: pick in #4026

**Review by XVilka — CHANGES_REQUESTED** [review]
Please rebase and fix test:
```
[XX] db/tools/rz_bin rz-bin -Pj
RZ_NOPLUGINS=1 /usr/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -eflirt.sigdb.load.system=false -eflirt.sigdb.load.home=false -N -Qc '!rz-bin -Pj ${RZ_FILE}~{}
echo
' bins/pdb/minimal.pdb
-- stdout
--- expected
+++ actual
@@ -0,6 +0,22 @@
 {
   "pdb": [
+    {
+      "modules": [
+        {
+          "module_name": "* Linker Generated Manifest RES *",
+          "object_file_name": ""
+        },
+        {
+          "module_name": "C:\\Users\\emuresan\\source\\repos\\minimal\\Release\\main.obj",
+          "object_file_name": "C:\\Users\\emuresan\\source\\repos\\minimal\\Release\\main.obj"
+        },
+        {
+          "module_name": "* Linker *",
+          "object_file_name": ""
+        }
+      ]
+    },
     {
       "types": [
         {

-- stderr
WARNING: core: cannot find base address, flags will probably be misplaced
WARNING: core: cannot find base address, flags will probably be misplaced
```

File: librz/bin/pdb/symbol.c (hunk 3 above)
Comment (b1llow) [reply librz/bin/pdb/symbol.c:25]: I have not found PDB files with this case.

**Review by XVilka — CHANGES_REQUESTED** [review]
```
FAILED: librz/bin/librz_bin.so.0.7.0.p/pdb_tpi.c.o 
gcc-12 -Ilibrz/bin/librz_bin.so.0.7.0.p -I. -I.. -Ilibrz -I../librz -Ilibrz/include -I../librz/include -I../librz/bin/format -I../subprojects/lz4-1.9.4/lib -Isubprojects/zstd-1.5.5/lib -I../subprojects/zstd-1.5.5/lib -I../subprojects/zstd-1.5.5/lib/common -I../subprojects/zstd-1.5.5/lib/compress -I../subprojects/zstd-1.5.5/lib/decompress -I../subprojects/zstd-1.5.5/lib/dictBuilder -I../subprojects/zstd-1.5.5/lib/legacy -Isubprojects/yxml -I../subprojects/yxml -Ilibrz/util/sdb/src -I../librz/util/sdb/src -I../librz/type/parser -I../subprojects/libmspack/libmspack/mspack -I../subprojects/libmspack/libmspack -fdiagnostics-color=always -fsanitize=address,undefined -fno-omit-frame-pointer -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -O2 -g -Wimplicit-fallthrough=3 -DRZ_PLUGIN_INCORE=1 -D_GNU_SOURCE --std=gnu99 -Werror=sizeof-pointer-memaccess -fvisibility=hidden -DASAN=1 -DRZ_ASSERT_STDOUT=1 -Wno-cpp -ftrivial-auto-var-init=pattern -funsigned-char -fPIC -DRZ_API_BIN_ONLY=1 -MD -MQ librz/bin/librz_bin.so.0.7.0.p/pdb_tpi.c.o -MF librz/bin/librz_bin.so.0.7.0.p/pdb_tpi.c.o.d -o librz/bin/librz_bin.so.0.7.0.p/pdb_tpi.c.o -c ../librz/bin/pdb/tpi.c
In function ‘tpi_data_free_with_kind’,
    inlined from ‘methodlist_parse’ at ../librz/bin/pdb/tpi.c:1234:2,
    inlined from ‘RzPdbTpiType_from_buf’ at ../librz/bin/pdb/tpi.c:1430:10:
../librz/bin/pdb/tpi.c:723:28: error: array subscript ‘Tpi_LF_Member[0]’ is partly outside array bounds of ‘Tpi_Type_MethodListMember[1]’ [-Werror=array-bounds]
  723 |                 free(lf_mem->name);
      |                      ~~~~~~^~~~~~
In file included from ../librz/include/rz_util.h:7,
                 from ../librz/include/rz_pdb.h:7,
                 from ../librz/bin/pdb/pdb.h:7,
                 from ../librz/bin/pdb/tpi.c:5:
In function ‘methodlist_parse’,
    inlined from ‘RzPdbTpiType_from_buf’ at ../librz/bin/pdb/tpi.c:1430:10:
../librz/include/rz_types.h:269:31: note: object of size 12 allocated by ‘calloc’
  269 | #define RZ_NEW0(x)       (x *)calloc(1, sizeof(x))
      |                               ^~~~~~~~~~~~~~~~~~~~
../librz/bin/pdb/tpi.c:1221:26: note: in expansion of macro ‘RZ_NEW0’
 1221 |                 member = RZ_NEW0(Tpi_Type_MethodListMember);
      |                          ^~~~~~~
../librz/include/rz_types.h:269:31: note: object of size 12 allocated by ‘calloc’
  269 | #define RZ_NEW0(x)       (x *)calloc(1, sizeof(x))
      |                               ^~~~~~~~~~~~~~~~~~~~
../librz/bin/pdb/tpi.c:1221:26: note: in expansion of macro ‘RZ_NEW0’
 1221 |                 member = RZ_NEW0(Tpi_Type_MethodListMember);
      |                          ^~~~~~~
../librz/include/rz_types.h:269:31: note: object of size 12 allocated by ‘calloc’
  269 | #define RZ_NEW0(x)       (x *)calloc(1, sizeof(x))
      |                               ^~~~~~~~~~~~~~~~~~~~
../librz/bin/pdb/tpi.c:1221:26: note: in expansion of macro ‘RZ_NEW0’
 1221 |                 member = RZ_NEW0(Tpi_Type_MethodListMember);
      |                          ^~~~~~~
../librz/include/rz_types.h:269:31: note: object of size 12 allocated by ‘calloc’
  269 | #define RZ_NEW0(x)       (x *)calloc(1, sizeof(x))
      |                               ^~~~~~~~~~~~~~~~~~~~
../librz/bin/pdb/tpi.c:1221:26: note: in expansion of macro ‘RZ_NEW0’
 1221 |                 member = RZ_NEW0(Tpi_Type_MethodListMember);
      |                          ^~~~~~~
../librz/include/rz_types.h:269:31: note: object of size 12 allocated by ‘calloc’
  269 | #define RZ_NEW0(x)       (x *)calloc(1, sizeof(x))
      |                               ^~~~~~~~~~~~~~~~~~~~
../librz/bin/pdb/tpi.c:1221:26: note: in expansion of macro ‘RZ_NEW0’
 1221 |                 member = RZ_NEW0(Tpi_Type_MethodListMember);
      |                          ^~~~~~~
cc1: all warnings being treated as errors
```

**Review by XVilka — APPROVED** [review]
